### PR TITLE
[election-ui-kotlin] - Work better with edge to edge enabled

### PR DESCRIPTION
**Pull request by samkavanaghbbc**
## What?
_Have the Kotlin apps work on Android 15 where edge to edge is enabled by default_

_Plus a few dependency updates_

## Why?

The UI was being drawn behind insets on the device such as camera / speaker.  These changes pad the content so the UI is only drawn in safe UI spaces.

Screenshots before changes (xml left, compose right):
Screenshot 2024-12-11 at 22 26 15 Screenshot 2024-12-11 at 22 27 28

After changes (xml left, compose right):
Screenshot 2024-12-11 at 22 28 32 Screenshot 2024-12-16 at 13 56 47

### CODE REVIEW

**Review by rt-bishop — APPROVED** [review]
All looks good to me. It's nice that only a few changes are required for a proper edge-to-edge support.

**Comment by aaronmoran** [comment]
Is there a reason for doing a fork rather than a branch?
Its not a problem, just a bit more awkward to run this as a reviewer

**Review by aaronmoran — CHANGES_REQUESTED** [review]
Doesn't work as expected on my phone - will send screenshot

**Comment by samkavanaghbbc** [comment]
> Is there a reason for doing a fork rather than a branch?
Its not a problem, just a bit more awkward to run this as a reviewer

Just that I don't have write permissions for the repo

File: election-ui-kotlin/build.gradle [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
+        material_version = '1.12.0'
     }
 }// Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id 'com.android.application' version '8.7.1' apply false
-    id 'com.android.library' version '8.7.1' apply false
+    id 'com.android.application' version '8.7.3' apply false
+    id 'com.android.library' version '8.7.3' apply false
     id 'org.jetbrains.kotlin.android' version "$kotlin_version" apply false
 }
```
Comment (aaronmoran) [inline election-ui-kotlin/build.gradle:26]: I'd recommend doing these dependency updates in a separate PR

**Review by aaronmoran — CHANGES_REQUESTED** [review]
refresh button doesn't work. Compose app Android api 33 phone

**Comment by aaronmoran** [comment]
> refresh button doesn't work. Compose app Android api 33 phone - its fine, emulator just wasn't running well

File: election-ui-kotlin/build.gradle (hunk 1 above)
Comment (samkavanaghbbc) [reply election-ui-kotlin/build.gradle:26]: Happy to if we think it'll cause any issues?  I don't know how many people have merge permissions for this repo (I don't), so was just keeping PRs to a minimum.

**Pull request merged**